src/main/java/net/syscon/elite/repository/jpa/model/CourtEvent.java

```diff
@@ -92,11 +99,30 @@
     @Column(name = "DIRECTION_CODE", length = 12)
     private String directionCode;
 
+    @OneToMany(mappedBy = "eventAndCharge.courtEvent", cascade = {CascadeType.REMOVE, CascadeType.PERSIST})
+    private final List<CourtEventCharge> charges = new ArrayList<>();
+
     public Optional<OffenderCourtCase> getOffenderCourtCase() {
         return Optional.ofNullable(offenderCourtCase);
     }
 
     public LocalDateTime getEventDateTime() {
         return eventDate.atTime(startTime.toLocalTime());
     }
+
+    @PrePersist
+    private void carryOverPreExistingChargesOnCreation() {
+        if (offenderCourtCase != null) {
+            log.debug("Adding pre-existing charges to new court event for court case '{}", offenderCourtCase.getId());
+
+            offenderCourtCase.getCharges().forEach(this::addPreExisting);
```

Comment:
So can the charges exist on several court events?

Reply:
Yes they can.  Bare in mind this is a separate entity to offender charges.  These are simply copies of the offender charges that exist on the court case for viewing in C-NOMIS.  I'll caveat this with this is my current understanding of NOMIS based on the conversations I have had to date and using C-NOMIS.

Reply:
An interesting data model!